test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/TagHelperDescriptorFactoryTest.cs

```diff
@@ -13,6 +17,161 @@ public class TagHelperDescriptorFactoryTest
+                        }
+                    },
+                    {
+                        typeof(MultiAttributeRequiredAttributeTagHelper),
+                        new[]
+                        {
+                            new TagHelperDescriptor(
+                                "div",
+                                typeof(MultiAttributeRequiredAttributeTagHelper).FullName,
+                                AssemblyName,
+                                attributes,
+                                requiredAttributes: new[] { "class" }),
+                            new TagHelperDescriptor(
+                                "input",
+                                typeof(MultiAttributeRequiredAttributeTagHelper).FullName,
+                                AssemblyName,
+                                attributes,
+                                requiredAttributes: new[] { "class" })
+                        }
+                    },
+                    {
+                        typeof(MultiRequiredAttributeTagHelper),
+                        new[]
+                        {
+                            new TagHelperDescriptor(
+                                "input",
+                                typeof(MultiRequiredAttributeTagHelper).FullName,
+                                AssemblyName,
+                                attributes,
+                                requiredAttributes: new[] { "class", "style" })
+                        }
+                    },
+                    {
+                        typeof(MultiTagMultiRequiredAttributeTagHelper),
+                        new[]
+                        {
+                            new TagHelperDescriptor(
+                                "div",
+                                typeof(MultiTagMultiRequiredAttributeTagHelper).FullName,
+                                AssemblyName,
+                                attributes,
+                                requiredAttributes: new[] { "class", "style" }),
+                            new TagHelperDescriptor(
+                                "input",
+                                typeof(MultiTagMultiRequiredAttributeTagHelper).FullName,
+                                AssemblyName,
+                                attributes,
+                                requiredAttributes: new[] { "class", "style" }),
+                        }
+                    },
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(AttributeTargetData))]
+        public void CreateDescriptor_UnderstandsAttributeTargets(
+            Type tagHelperType,
+            IEnumerable<TagHelperDescriptor> expectedDescriptors)
+        {
+            // Act
+            var descriptors = TagHelperDescriptorFactory.CreateDescriptors(
+                AssemblyName,
+                tagHelperType,
+                new ParserErrorSink());
```

Comment:
suggest these tests should confirm the `ParserErrorSink` contains no errors after the Act